### PR TITLE
[GEOT-6389] SLD vendor option graphic-margin can't be empty

### PR DESCRIPTION
**Pull request by dfiorent**
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by dfiorent** [comment]
Discard empty SLD vendor option tags in SLD.

File: modules/library/xml/src/main/java/org/geotools/xml/styling/SLDParser.java [hunk 1]
```diff
@@ -1253,8 +1253,9 @@ protected OtherText parseOtherText(Node root) {
     private void parseVendorOption(Map<String, String> options, Node child) {
         String key = child.getAttributes().getNamedItem("name").getNodeValue();
         String value = getFirstChildValue(child);
-
-        options.put(key, value);
+        if (value != null) {
```
Comment (aaime) [inline modules/library/xml/src/main/java/org/geotools/xml/styling/SLDParser.java:1256]: I would go and make sure it's not null, not empty, and not blank, all vendor options are supposed to have some content.
Commons-lang3 is in the classpath here, using ``SystemUtils.isNotBlank`` should do the trick.

File: modules/library/xml/src/test/java/org/geotools/xml/styling/SLDParserTest.java [hunk 2]
```diff
@@ -630,7 +655,8 @@ public void testExternalEntitiesDisabled() {
         // this SLD file references as external entity a file on the local filesystem
         SLDParser parser = new SLDParser(styleFactory, input(SLD_EXTERNALENTITY));
 
-        // without a custom EntityResolver, the parser will try to read the entity file on the local
+        // without a custom EntityResolver, the parser will try to read the entity file
```
Comment (aaime) [inline modules/library/xml/src/test/java/org/geotools/xml/styling/SLDParserTest.java:658]: Reformats in PRs are not welcomed, please remove all of them.

File: modules/library/xml/src/test/java/org/geotools/xml/styling/SLDParserTest.java [hunk 3]
```diff
@@ -756,6 +784,20 @@ public void testContrastEnhancement() throws Exception {
...
+
+        SLDParser parser = new SLDParser(styleFactory, input(SLD_NULL_VENDOR_OPTION));
+        Style[] styles = parser.readXML();
+        List<FeatureTypeStyle> fts = styles[0].featureTypeStyles();
+        List<Rule> rules = fts.get(0).rules();
+        List<Symbolizer> symbolizers = rules.get(0).symbolizers();
+
+        PointSymbolizer ps = (PointSymbolizer) symbolizers.get(0);
+
+        assertEquals(1, ps.getOptions().size());
```
Comment (aaime) [inline modules/library/xml/src/test/java/org/geotools/xml/styling/SLDParserTest.java:798]: Also make sure it's the one you want here (by checking the available keys).

**Comment by aaime** [comment]
QA Travis check fails with:

```
[ERROR] Found 1 non-complying files, failing build
[ERROR] To fix formatting errors, run "mvn fmt:format"
[ERROR] Non complying file: /home/travis/build/geotools/geotools/modules/library/xml/src/test/java/org/geotools/xml/styling/SLDParserTest.java
```

**Comment by aaime** [comment]
This PR was done on GeoSolutions's behalf, and is covered by our CLA. Merging.

**Pull request merged**